Wikipedia/Code/StorageAndSyncingSettingsViewController.swift

```diff
@@ -0,0 +1,213 @@
+struct SettingsSection {
+    let headerTitle: String?
+    let footerText: String?
+    let items: [SettingsItem]
+}
+
+struct SettingsItem {
+    let disclosureType: WMFSettingsMenuItemDisclosureType?
+    let itemType: StorageAndSyncingSettingsItemType
+    let title: String?
+    let iconName: String?
+    let iconColor: UIColor?
+    let iconBackgroundColor: UIColor?
+    let isSwitchOn: Bool?
+    let buttonTitle: String?
+}
+
+enum StorageAndSyncingSettingsItemType: Int {
+    case syncSavedArticlesAndLists, showSavedReadingList, eraseSavedArticles, syncWithTheServer
+    
+    public func settingsItem(isSwitchOnByDefault: Bool? = nil) -> SettingsItem {
+        var disclosureType: WMFSettingsMenuItemDisclosureType? = nil
+        var title: String? = nil
+        var isSwitchOn: Bool? = nil
+        var buttonTitle: String? = nil
+        switch self {
+        case .syncSavedArticlesAndLists:
+            disclosureType = .switch
+            title = WMFLocalizedString("settings-storage-and-syncing-enable-sync-title", value: "Sync saved articles and lists", comment: "Title of the settings option that enables saved articles and reading lists syncing")
+            isSwitchOn = isSwitchOnByDefault
+        case .showSavedReadingList:
+            disclosureType = .switch
+            title = WMFLocalizedString("settings-storage-and-syncing-show-default-reading-list-title", value: "Show Saved reading list", comment: "Title of the settings option that enables showing the default reading list")
+            isSwitchOn = isSwitchOnByDefault
+        case .syncWithTheServer:
+            disclosureType = .titleButton
+            buttonTitle = WMFLocalizedString("settings-storage-and-syncing-server-sync-title", value: "Sync with the server", comment: "Title of the settings button that initiates saved articles and reading lists server sync")
+        default:
+            break
+        }
+        
+        return SettingsItem(disclosureType: disclosureType, itemType: self, title: title, iconName: nil, iconColor: nil, iconBackgroundColor: nil, isSwitchOn: isSwitchOn, buttonTitle: buttonTitle)
+    }
+}
+
+@objc(WMFStorageAndSyncingSettingsViewController)
+class StorageAndSyncingSettingsViewController: UIViewController {
+    private let customViewCellReuseIdentifier = "CustomViewTableViewCell"
+    private var theme: Theme = Theme.standard
+    private var tableView: UITableView!
```

Comment:
Using an IBOutlet & storyboard(or xib) for the `tableView` would eliminate the bang here and the boilerplate tableview config code in `viewDidLoad`.

Reply:
(I don't feel super strongly about this, but it would be nice.)